nginx/etc/nginx/templates/conf.d/site.conf

```diff
@@ -0,0 +1,45 @@
+{{ $acme_domain := env "ACME_DOMAIN" }}
+{{ $ssl_ready := env "SSL_READY" }}
+
+# If we're listening on https, define an http listener that redirects everything to https
+{{ if eq $ssl_ready "true" }}
+server { 
+    server_name _;
+    listen      80;
+
+    include /etc/nginx/health.conf;
+    
+    location / {
+        return 301 https://$host$request_uri;
+    }
+}
+{{ end }}
+
+# The main server block
+server {
+    server_name _;
+    # Listen on port 80 unless we have certificates installed, then listen on 443
+    listen {{ if ne $ssl_ready "true" }}80{{ else }}443 ssl{{ end }};
+
+    include /etc/nginx/health.conf;
+
+    location /.well-known/acme-challenge {
+        alias /var/www/acme/challenge;
+    }
+
+    {{ if service "wordpress" }}
```

Comment:
This block inside `{{ if service "wordpress" }}` is the only block that's needed here. Everything else is likely to be repeated for other consumers of the Nginx image.